crates/sui-oracle/move/oracle/sources/simple_oracle.move

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module oracle::simple_oracle {
+    use std::option::{Self, Option};
+    use std::string;
+    use std::string::String;
+
+    use oracle::data::{Self, Data};
+    use sui::clock::{Self, Clock};
+    use sui::dynamic_field as df;
+    use sui::object::{Self, UID};
+    use sui::table;
+    use sui::table::Table;
+    use sui::transfer;
+    use sui::tx_context::{Self, TxContext};
+
+    const ESenderNotOracle: u64 = 0;
+    const ETickerNotExists: u64 = 1;
+
+    struct SimpleOracle has store, key {
+        id: UID,
+        /// The address of the oracle.
+        address: address,
+        /// The name of the oracle.
+        name: String,
+        /// The description of the oracle.
+        description: String,
+        /// The URL of the oracle.
+        url: String,
+    }
+
+    struct StoredData<T: store> has copy, store, drop {
+        value: T,
+        sequence_number: u64,
+        timestamp: u64,
+        /// An identifier for the reading (for example real time of observation, or sequence number of observation on other chain).
+        identifier: String,
+    }
+
+    public fun get_historical_data<K: copy + drop + store, V: store + copy>(
+        oracle: &SimpleOracle,
+        ticker: String,
+        archival_key: K
+    ): Option<Data<V>> {
+        string::append(&mut string::utf8(b"[historical] "), ticker);
+        let historical_data: &Table<K, StoredData<V>> = df::borrow(&oracle.id, ticker);
+        let StoredData { value, sequence_number, timestamp, identifier } = *table::borrow(
+            historical_data,
+            archival_key
+        );
+        option::some(data::new(value, ticker, sequence_number, timestamp, oracle.address, identifier))
+    }
+
+    public fun get_latest_data<T: store + copy>(oracle: &SimpleOracle, ticker: String): Option<Data<T>> {
+        if (!df::exists_(&oracle.id, ticker)) {
+            return option::none()
+        };
+        let data: &StoredData<T> = df::borrow(&oracle.id, ticker);
+        let StoredData { value, sequence_number, timestamp, identifier } = *data;
+        option::some(data::new(value, ticker, sequence_number, timestamp, oracle.address, identifier))
+    }
+
+    /// Create a new shared SimpleOracle object for publishing data.
+    public entry fun create(name: String, url: String, description: String, ctx: &mut TxContext) {
+        let oracle = SimpleOracle { id: object::new(ctx), address: tx_context::sender(ctx), name, description, url };
```

Comment:
It seems here that we only authenticate by address, should we provide a mode where we authenticate by a capability? This allows oracles more flexible security policies when updating.